firstapp/app/src/androidTest/java/com/example/firstapp/UtilityMethods.java

```diff
@@ -26,13 +36,28 @@ static UiDevice launchApp(String pkg, int timeout) {
         // Launch the app
         Context context = ApplicationProvider.getApplicationContext();
         final Intent intent = context.getPackageManager().getLaunchIntentForPackage(pkg);
+
         // Clear out any previous instances
         intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK);
+
+        //startActivity imposes a 5 second cool-down after the home button is pressed, so we wait
+        //out that cool-down before grabbing the time and launching
```

Comment:
Who is imposing that cool-down?  If external, any documentation?

Reply:
haven't seen any documentation for it but I've observed and it seems like others have too. https://stackoverflow.com/questions/5600084/starting-an-activity-from-a-service-after-home-button-pressed-without-the-5-seco

Reply:
Thanks.  May want to avoid the need to press the home button altogether.

Reply:
Interesting. Looks like android does a weird thing where it doesn't physically close an app when you force quit it and you open another app immediately afterward (even if you sleep in between). Thus, not pressing home button might be problematic when doing iterations if we want a clean visual on startup. Something to consider down the line though.

Reply:
force-quit should kill the app.  However, the recent-apps may still show it, which is a bug with recent-apps itself.

Reply:
I mean that when you force-quit an app only to open up another app in the same test, instead of closing the app, going to the launcher, and reopening it, the new app gets swiped in from the right side of the screen. This isn't really problematic and might be worth doing to save the 5 seconds per test.

Reply:
I think what is happening is that the force-quit command didn't really finish.

So you'd want something like the following:

    Process process = new ProcessBuilder(...).start();
    process.waitFor(1000, TimeUnit.MILLISECONDS);

Reply:
same thing happens if I add this line (and even if I wait for 5000 millis instead)

Reply:
Ahh, so it might be an issue with startActivity.  The following shouldn't have any issue:

    adb shell am force-stop com.google.android.apps.maps
    adb shell am start com.google.android.apps.maps

Reply:
im not able to get start to work, even from the cmd line

Reply:
What is the exact command that you used?  It sounds like you need to pass in your activity name:

    adb shell am start com.package.name/.MainActivity

Reply:
Would I just assume that every app names their opening page "MainActivity"?